File: 2015/day2/part1.py
def part1(dimensions):
    ret = 0
    for line in dimensions.split('\n'):
        line_split = list(map(int, line.split('x')))
        ret += 2 * line_split[0] * line_split[1] \
            + 2 * line_split[1] * line_split[2] \
            + 2 * line_split[0] * line_split[2] \
            + min(line_split)
    return ret

File: 2015/day2/part1.swift
import Foundation

func part1(_ input: String) -> Int {
	let inputSplit = input.components(separatedBy: "\n")
	let result = inputSplit.map { (line: String) -> Int in
		let dimensions = line.components(separatedBy: "x").map { Int(String($0)) ?? 0 }.sorted()
		// surface area (2 * l * w + 2 * w * h + 2 * h * l) + area of smallest side
		return 2 * dimensions[0] * dimensions[1] +
			2 * dimensions[1] * dimensions[2] +
			2 * dimensions[2] * dimensions[0] +
			dimensions[0] * dimensions[1]
	}.reduce(0, +)
	return result
}